From 18b7c997ac46f6ca8f65ab79d7ee84a055528639 Mon Sep 17 00:00:00 2001
From: minnnyres <liuthgw23@gmail.com>
Date: Mon, 18 Dec 2023 12:50:57 +0000
Subject: [PATCH 2/2] support ffmpeg arm64 mingw

---
 ports/ffmpeg/portfile.cmake | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/ports/ffmpeg/portfile.cmake b/ports/ffmpeg/portfile.cmake
index 9852edd0c..9a28fc24b 100644
--- a/ports/ffmpeg/portfile.cmake
+++ b/ports/ffmpeg/portfile.cmake
@@ -31,7 +31,7 @@ if(NOT VCPKG_TARGET_ARCHITECTURE STREQUAL "wasm32")
     vcpkg_add_to_path("${NASM_EXE_PATH}")
 endif()
 
-set(OPTIONS "--enable-pic --disable-doc --enable-debug --enable-runtime-cpudetect --disable-autodetect")
+set(OPTIONS "--enable-pic --disable-doc --disable-debug --enable-runtime-cpudetect --disable-autodetect")
 
 if(VCPKG_HOST_IS_WINDOWS)
     vcpkg_acquire_msys(MSYS_ROOT PACKAGES automake1.16)
@@ -42,11 +42,17 @@ else()
     find_program(SHELL bash)
 endif()
 
+if(VCPKG_HOST_IS_LINUX)
+    string(APPEND OPTIONS " --pkg-config=/usr/bin/pkg-config")
+endif()
+
 if(VCPKG_TARGET_IS_MINGW)
     if(VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
         string(APPEND OPTIONS " --target-os=mingw32")
     elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
         string(APPEND OPTIONS " --target-os=mingw64")
+    elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")
+        string(APPEND OPTIONS " --target-os=mingw64")
     endif()
 elseif(VCPKG_TARGET_IS_LINUX)
     string(APPEND OPTIONS " --target-os=linux --enable-pthreads")
-- 
2.39.2

