From 5f6f7785cb46d86d2531a776bf75ee105522d05d Mon Sep 17 00:00:00 2001
From: minnnyres <liuthgw23@gmail.com>
Date: Wed, 27 Dec 2023 10:59:04 +0000
Subject: [PATCH 4/4] build ffmpeg with d3d11va and dxva2

---
 ports/ffmpeg/portfile.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ports/ffmpeg/portfile.cmake b/ports/ffmpeg/portfile.cmake
index 9a28fc24b..cdf90df1e 100644
--- a/ports/ffmpeg/portfile.cmake
+++ b/ports/ffmpeg/portfile.cmake
@@ -54,6 +54,7 @@ if(VCPKG_TARGET_IS_MINGW)
     elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")
         string(APPEND OPTIONS " --target-os=mingw64")
     endif()
+    string(APPEND OPTIONS " --enable-d3d11va --enable-dxva2 --enable-mediafoundation")
 elseif(VCPKG_TARGET_IS_LINUX)
     string(APPEND OPTIONS " --target-os=linux --enable-pthreads")
 elseif(VCPKG_TARGET_IS_UWP)
-- 
2.39.2

